File: wwwroot/js/seguimientoactivo.js
// JavaScript para Seguimiento Activo
// Patrón DataTables server-side (igual que Clientes Leads)

let tablaSeguimiento;

function obtenerFiltrosSeguimiento() {
    return {
        filtroCliente: $('#filtroCliente').val() || '',
        filtroAgente: $('#filtroAgente').val() || '',
        filtroTipoAccion: $('#filtroTipoAccion').val() || '',
        filtroEstado: $('#filtroEstado').val() || ''
    };
}

function cargarFiltrosSeguimiento() {
    $.get('/SeguimientoActivo/Clientes', function (response) {
        if (response.success && response.data) {
            var select = $('#filtroCliente');
            select.find('option:not(:first)').remove();
            response.data.forEach(function (id) {
                select.append($('<option>', { value: id, text: id }));
            });
        }
    });

    $.get('/SeguimientoActivo/Agentes', function (response) {
        if (response.success && response.data) {
            var select = $('#filtroAgente');
            select.find('option:not(:first)').remove();
            response.data.forEach(function (agente) {
                select.append($('<option>', { value: agente, text: agente }));
            });
        }
    });
}

function inicializarDataTablesSeguimiento() {
    console.log('📊 Inicializando DataTables Seguimiento Activo...');

    tablaSeguimiento = $('#tablaSeguimiento').DataTable(DataTablesCommon.baseConfig({
        order: [[2, 'desc']],
        pageLength: 25,
        ajax: DataTablesCommon.ajaxPost('/SeguimientoActivo/GetData', function (d) {
            Object.assign(d, obtenerFiltrosSeguimiento());
        }),
        columns: [
            { data: 'iD_Cliente', name: 'ID_Cliente' },
            { data: 'agente', name: 'Agente' },
            {
                data: 'fecha_Accion',
                name: 'Fecha_Accion',
                render: function (data, type) {
                    if (type === 'display' || type === 'filter') {
                        return formatearFecha(data);
                    }
                    return data;
                }
            },
            { data: 'tipo_Accion', name: 'Tipo_Accion' },
            { data: 'descripcion_Accion', name: 'Descripcion_Accion' },
            { data: 'resultado', name: 'Resultado' },
            {
                data: 'estado',
                name: 'Estado',
                render: function (data) {
                    var badgeEstado = data === 'Completado' ? 'badge-completado' : 'badge-pendiente';
                    return '<span class="badge ' + badgeEstado + '">' + (data || '') + '</span>';
                }
            },
            {
                data: 'fecha_Proximo_Contacto',
                name: 'Fecha_Proximo_Contacto',
                render: function (data, type) {
                    if (type === 'display' || type === 'filter') {
                        return formatearFecha(data);
                    }
                    return data;
                }
            }
        ]
    }));

    console.log('✅ DataTables inicializado');
}

function recargarTablaSeguimiento() {
    if (tablaSeguimiento) {
        tablaSeguimiento.ajax.reload();
    }
}

function formatearFecha(fecha) {
    if (!fecha) return '-';
    var d = new Date(fecha);
    return d.toLocaleDateString('es-ES');
}

$(document).ready(function () {
    cargarFiltrosSeguimiento();

    if (typeof inicializarDataTablesSeguimiento === 'function') {
        inicializarDataTablesSeguimiento();
    }

    $('#btnBuscar').on('click', function () {
        recargarTablaSeguimiento();
    });

    $('#filtroCliente, #filtroAgente, #filtroTipoAccion, #filtroEstado').on('change', function () {
        recargarTablaSeguimiento();
    });
});

File: DTOs/SeguimientoActivo.cs
namespace Inmobiliaria.Net8.DTOs
{
    /// <summary>
    /// DTO para Seguimiento Activo
    /// Representa acciones de seguimiento realizadas sobre clientes leads
    /// </summary>
    public class SeguimientoActivo
    {
        public string ID_Cliente { get; set; } = string.Empty;
        public string? ID_Propiedad { get; set; }
        public string Agente { get; set; } = string.Empty;
        public string? Codigo_Agente { get; set; }
        public DateTime Fecha_Accion { get; set; }
        public string Tipo_Accion { get; set; } = string.Empty;
        public string Descripcion_Accion { get; set; } = string.Empty;
        public string Resultado { get; set; } = string.Empty;
        public string Estado { get; set; } = string.Empty;
        public DateTime? Fecha_Proximo_Contacto { get; set; }
        
        // Para DataTables
        public int? TotalRowCount { get; set; }
    }

    /// <summary>
    /// DTO para filtros de búsqueda de Seguimiento Activo
    /// </summary>
    public class FiltroSeguimientoActivo
    {
        public string? Agente { get; set; }
        public string? ID_Cliente { get; set; }
        public string? Tipo_Accion { get; set; }
        public DateTime? FechaDesde { get; set; }
        public DateTime? FechaHasta { get; set; }
        public string? ID_Propiedad { get; set; }
        public string? Estado { get; set; }
        public int PageNumber { get; set; } = 1;
        public int PageSize { get; set; } = 25;
        public string ColumnaOrden { get; set; } = "Fecha_Accion";
        public string DireccionOrden { get; set; } = "DESC";
    }
}

